### ISSUE TITLE
Incorrect pattern matching behaviour when matching tuples of arbitrary length

### ISSUE THREAD

**Issue by ImogenBits** [microsoft/pyright#7826]
When you use pattern matching against a tuple of arbitrary length pyright seems to think that an empty sequence pattern will always match. Strangely enough, the type inside the case statement is inferred correctly as a length 0 tuple, but all following cases are narrowed to Never.

E.g. in this function
```py
def f(thingy: tuple[int, ...]) -> int:
    match thingy:
        case []:
            return 1
        case [_]:
            return 2
        case _:
            return 3
```
pyright says that the second and third cases are never matched since the subject type is Never. If you swap the first and second cases the [_] case now works properly, but the wildcard pattern still supposedly never matches.

I'm using the VSCode extension with pyright version 1.1.360.

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.362, which I just published.
